corehq/apps/app_manager/util.py

```diff
@@ -649,11 +649,15 @@ def get_latest_enabled_app_release(domain, location_id, app_id):
     location = SQLLocation.active_objects.get(location_id=location_id)
     location_and_ancestor_ids = location.get_ancestors(include_self=True).values_list(
         'location_id', flat=True).reverse()
+    # get all active enabled releases and order by version desc to get one with the highest version in the end
+    # for a location. Do not use the first object itself in order to respect the location hierarchy and use
+    # the closest location to determine the valid active release
     latest_enabled_releases = {
         enabled_release.location_id: enabled_release.build_id
         for enabled_release in
         LatestEnabledAppRelease.objects.filter(
-            location_id__in=location_and_ancestor_ids, app_id=app_id, domain=domain, active=True)
+            location_id__in=location_and_ancestor_ids, app_id=app_id, domain=domain, active=True).order_by(
+            'version')
```

Comment:
You might be able to cut down results returned here with a `.distinct()`, though I'm not certain what that'd look like

Reply:
not sure how that would help. we need all values set for a location.

Reply:
what for?  Don't you only need the latest one?  The dict comprehension only preserves one value per location, the others are thrown away.

Reply:
offlined. distinct would still keep all since version would be different.